docs: upgrading notes (postgres archive and `ARGO_TEMPLATE`) (#13803)
argoproj/argo-workflows: docs/upgrading.md

When returning a map or array in an expression, you would get a Golang representation.
This now returns plain JSON.

### `ARGO_TEMPLATE` removed from main container

> The environment variable `ARGO_TEMPLATE` which is an internal implementation detail is no longer available inside the `main` container of your workflow pods.
> This is documented here as we are aware that some users of Argo Workflows use this.

Reviewer: ```suggestion
The environment variable `ARGO_TEMPLATE` is no longer available inside the `main` container of your workflow pods.
This is an internal implementation detail, but is documented here as we are aware that some Argo users have utilized this.
```

- Clarify by modifying the sentence structure a bit - 1 sentence for the change, the other sentence for the internal piece
  -  also, without commas surrounding it, "which is an internal implementation detail" is grammatically incorrect
- The docs consistently refer to "Argo Workflows" as "Argo", which I've noted elsewhere too
